src/lib/form-field/_form-field-theme.scss

```diff
@@ -198,6 +206,21 @@
       @include _mat-form-field-placeholder-float-nodedupe(
               $subscript-font-scale, $infix-padding, $infix-margin-top);
     }
+
+    // Server-side rendered matInput with focus (used as a pure CSS stand-in for
+    // mat-form-field-should-float).
+    .mat-input-server:focus + .mat-form-field-placeholder-wrapper .mat-form-field-placeholder {
+      @include _mat-form-field-placeholder-floating(
+              $subscript-font-scale, $infix-padding, $infix-margin-top);
+    }
```

Comment:
Can't this one be combined with the other one above?
(`&.mat-form-field-should-float .mat-form-field-placeholder`)